resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -422,6 +446,22 @@ private[yarn] class YarnAllocator(
     }
   }
 
+  private def handleNodesInDecommissioningState(allocateResponse: AllocateResponse): Unit = {
+    // Some of the nodes are put in decommissioning state where RM did allocate
+    // resources on those nodes for earlier allocateResource calls, so notifying driver
+    // to put those executors in decommissioning state
+    allocateResponse.getUpdatedNodes.asScala.filter (node =>
+      node.getNodeState == NodeState.DECOMMISSIONING &&
```

Comment:
So I'm seeing this break the the hadoop-2.7 profile build.  NodeState.DECOMMISSIONING was added in hadoop 2.8 https://issues.apache.org/jira/browse/YARN-41
@abhishekd0907 @mridulm

Reply:
fyi filed https://issues.apache.org/jira/browse/SPARK-39491

Reply:
@tgravescs 
i have created a PR for this, please review
https://github.com/apache/spark/pull/36890